src/sync/sync_session.cpp

```diff
@@ -473,18 +474,22 @@ void SyncSession::handle_progress_update(uint64_t downloaded, uint64_t downloada
     }
 }
 
-void SyncSession::NotifierPackage::update(const Progress& current_progress)
+void SyncSession::NotifierPackage::update(const Progress& current_progress, bool data_is_fresh)
 {
-    if (is_streaming || captured_transferrable)
+    if (is_streaming || captured_transferrable || !data_is_fresh)
         return;
 
     captured_transferrable = direction == NotifierType::download ? current_progress.downloadable
                                                                  : current_progress.uploadable;
 }
 
-std::function<void()> SyncSession::NotifierPackage::create_invocation(const Progress& current_progress, bool& is_expired) const
+std::function<void()> SyncSession::NotifierPackage::create_invocation(const Progress& current_progress,
+                                                                      bool& is_expired) const
 {
-    REALM_ASSERT(is_streaming || captured_transferrable);
+    // It's possible for a non-streaming notifier to not yet have fresh transferrable bytes data.
+    // In that case, we don't call it at all.
```

Comment:
That much I understand. But this seems to result in the block _never_ being called, rather than waiting for concrete information about the progress to be available?